streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -77,6 +72,59 @@ public int compareTo(AssignedPartition that) {
         }
     }
 
+    private static class ClientMetadata {
+        final HostInfo hostInfo;
+        final Set<String> consumers;
+        ClientState<TaskId> state;
+
+        ClientMetadata(String endPoint) {
+
+            // get the host info if possible
+            if (endPoint != null) {
```

Comment:
Users may choose not to provide this config in `StreamsConfig`, if they do not need the IQ feature.